snapcraft/internal/meta/_desktop.py

```diff
@@ -35,21 +35,17 @@ def __init__(
         self._prime_dir = prime_dir
         self._path = os.path.join(prime_dir, filename)
         if not os.path.exists(self._path):
-            raise errors.InvalidDesktopFileError(
-                filename, 'does not exist (defined in the app {!r})'.format(
-                    name))
+            raise errors.MissingDesktopFileError(filename,app_name)
```

Comment:
Okay, now things are starting to come together. Please separate these two parameters with a space after the comma, though, to help readability.